web/packages/teleport/src/lib/useMfa.ts

```diff
@@ -77,27 +99,67 @@ export function useMfa(emitterSender: EventEmitterMfaSender): MfaState {
       });
   }
 
-  const onChallenge = useCallback(challengeJson => {
-    const { webauthnPublicKey, ssoChallenge, totpChallenge } =
-      makeMfaAuthenticateChallenge(challengeJson);
-
-    setState(prevState => ({
-      ...prevState,
-      ssoChallenge,
-      webauthnPublicKey,
-      totpChallenge,
-    }));
-  }, []);
+  const waitForSsoChallengeResponse = useCallback(
+    async (
+      ssoChallenge: SSOChallenge,
+      abortSignal: AbortSignal
+    ): Promise<void> => {
+      const channel = new BroadcastChannel(ssoChallenge.channelId);
+
+      try {
+        const event = await waitForMessage(channel, abortSignal);
+        emitterSender.sendChallengeResponse({
+          sso_response: {
+            requestId: ssoChallenge.requestId,
+            token: event.data.mfaToken,
+          },
+        });
+        clearChallenges();
+      } catch (error) {
+        if (error.name !== 'AbortError') {
+          throw error;
+        }
+      } finally {
+        channel.close();
+      }
+    },
+    [emitterSender]
+  );
 
   useEffect(() => {
-    if (emitterSender) {
-      emitterSender.on(TermEvent.WEBAUTHN_CHALLENGE, onChallenge);
-
-      return () => {
-        emitterSender.removeListener(TermEvent.WEBAUTHN_CHALLENGE, onChallenge);
-      };
-    }
-  }, [emitterSender, onChallenge]);
+    let ssoChallengeAbortController: AbortController | undefined;
+    const chanllengeHandler = (challengeJson: string) => {
+      const { webauthnPublicKey, ssoChallenge, totpChallenge } =
+        makeMfaAuthenticateChallenge(challengeJson);
+
+      setState(prevState => ({
+        ...prevState,
+        addMfaToScpUrls: true,
+        ssoChallenge,
+        webauthnPublicKey,
+        totpChallenge,
+      }));
+
+      if (ssoChallenge) {
+        ssoChallengeAbortController?.abort();
+        ssoChallengeAbortController = new AbortController();
+        void waitForSsoChallengeResponse(
+          ssoChallenge,
+          ssoChallengeAbortController.signal
+        );
+      }
+    };
+
+    emitterSender?.on(TermEvent.WEBAUTHN_CHALLENGE, chanllengeHandler);
```

Comment:
nit: We should rename this const to `MFA` as I did on the backend.
```suggestion
    emitterSender?.on(TermEvent.MFA_CHALLENGE, chanllengeHandler);
```